### PR TITLE
Type definitions for NodeJS 14 and Mocha 8.4

### PR DESCRIPTION
**Pull request by jozefizso**
Changes:

* Use @types/node definitions for Node.js v14
* Unify mocha package to v8.4.0

### CODE REVIEW

File: packages/custom-functions-metadata/test/src/testCases.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ describe("test cases", function() {
 
                         // if actual errors are different than expected, write out the actual errors to a file
                         // otherwise, delete the actual errors file if it exists
-                        if (actualErrors !== expectedErrors) {
+                        if (actualErrors && actualErrors !== expectedErrors) {
```
Comment (igor-ribeiiro) [inline packages/custom-functions-metadata/test/src/testCases.ts:73]: @jozefizso What is the reason for this change?

Comment (akrantz) [inline packages/custom-functions-metadata/test/src/testCases.ts:73]: Curious, why is this change needed?

**Review by akrantz — COMMENTED** [review]
Thanks for the pull request!  We appreciate the help in keeping things up-to-date.

File: packages/custom-functions-metadata/test/src/testCases.ts (hunk 1 above)
Comment (jozefizso) [reply packages/custom-functions-metadata/test/src/testCases.ts:73]: The definition of `fs.writeFileSync()` changed in NodeJS 14 and it does not allow `undefined` value.

Without the change the test will fail to compile:

```
TSError: ⨯ Unable to compile TypeScript:
test/src/testCases.ts:74:64 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string | Uint8Array | Uint8ClampedArray | Uint16Array | Uint32Array | Int8Array | Int16Array | ... 5 more ... | DataView'.
  Type 'undefined' is not assignable to type 'string | Uint8Array | Uint8ClampedArray | Uint16Array | Uint32Array | Int8Array | Int16Array | ... 5 more ... | DataView'.

74                             fs.writeFileSync(actualErrorsFile, actualErrors);
                                                                  ~~~~~~~~~~~~
```

I think guarding against the `undefined` value is correct change here.

Comment (jozefizso) [reply packages/custom-functions-metadata/test/src/testCases.ts:73]: See comment above.

**Comment by igor-ribeiiro** [comment]
Hey @jozefizso thanks for the PR. Can you also upload the `.lock` files? This is the error our CI is throwing:
```
lerna ERR! npm ci exited 1 in 'custom-functions-metadata'
lerna ERR! npm ci stderr:
npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
npm ERR! 
npm ERR! 
npm ERR! Invalid: lock file's @types/node@12.12.34 does not satisfy @types/node@^14.17.2
npm ERR! Invalid: lock file's @types/node-fetch@2.5.5 does not satisfy @types/node-fetch@^2.5.10
npm ERR! Invalid: lock file's mocha@7.1.1 does not satisfy mocha@^8.4.0
npm ERR!
```

**Comment by igor-ribeiiro** [comment]
@jozefizso still giving similar `.lock` errors. I think you need to commit all the `.lock` files from all the packages you changed in this PR

**Comment by jozefizso** [comment]
2nd try with the lock files :)

File: packages/custom-functions-metadata/test/src/testCases.ts (hunk 1 above)
Comment (akrantz) [reply packages/custom-functions-metadata/test/src/testCases.ts:73]: Let me look over the code. I think we might want to do the undefined check in a different way...

Comment (akrantz) [reply packages/custom-functions-metadata/test/src/testCases.ts:73]: I think it is OK.

**Review by akrantz — APPROVED** [review]

**Comment by jozefizso** [comment]
Thank you.
